### ISSUE TITLE
[BUG] scroll to top button is displayed when scrollview changes size

### ISSUE THREAD

**Issue by KristianBalaj** [NamanShergill/flutter_scroll_to_top#3]
**Actual behavior:**
Having an infinite scroll list (items loaded on scrolling) when I scroll to top at the same time items are being loaded. Then the scroll to top button remains displays when the items get loaded (the scroll view changes height).

**Expected behavior:**
The scroll to top button should not display when the scroll view changes its size e.g. height.

**Comment by clragon** [comment]
I am also having this issue right now.

> Unfortunately the library can't use the scroll controller implementation because that means it stops working for scrollables in the body of `NestedScrollView`

That would be quite the problem indeed.
But we can still make this work. I propse the following solution:

```dart
double offset = widget.scrollController?.offset ?? notification.metrics.pixels;

if (offset  > widget.promptScrollOffset &&  !scrollTopAtOffset) {
  // stuff
} else if (offset <= widget.promptScrollOffset && scrollTopAtOffset) {
  // other stuff
}
```

**Comment by clragon** [comment]
I'm definitely also having this problem right now.
I can't come up with a way we could fix this though.

Maybe we could simply add an optional parameter that dictates these behaviours?

**Comment by NamanShergill** [comment]
> I'm definitely also having this problem right now.
> I can't come up with a way we could fix this though.
> 
> Maybe we could simply add an optional parameter that dictates these behaviours?

Seems like the best approach, a bool to use the scroll controller instead of the current implementation which would be set to false by default.

**Comment by NamanShergill** [comment]
Fixed in 2.0.0
